packages/stores/src/queries-external/token-pair-historical-chart/index.ts

```diff
@@ -0,0 +1,93 @@
+import { KVStore } from "@keplr-wallet/common";
+import { HasMapStore } from "@keplr-wallet/stores";
+import { computed, makeObservable } from "mobx";
+import { IPriceStore } from "src/price";
+
+import { IMPERATOR_HISTORICAL_DATA_BASEURL } from "..";
+import { ObservableQueryExternalBase } from "../base";
+import { PriceRange, TokenPairHistoricalPrice } from "./types";
+
+const AvailableRangeValues = ["7d", "1mo", "1y"] as const;
+type Tf = PriceRange;
+
+/** Queries Imperator token history data chart. */
+export class ObservableQueryTokenPairHistoricalChart extends ObservableQueryExternalBase<
+  TokenPairHistoricalPrice[]
+> {
+  constructor(
+    kvStore: KVStore,
+    baseURL: string,
+    protected readonly priceStore: IPriceStore,
+    protected readonly poolId: string,
+    protected readonly baseDenom: string,
+    protected readonly quoteDenom: string,
+    /**
+     * Range of historical data
+     * Available values: ["7d", "1mo", "1y"]
+     */
+    protected readonly tf: Tf
+  ) {
+    super(
+      kvStore,
+      "https://api.osmosis.zone/",
+      // baseURL,
+      `/pairs/v1/historical/2/chart?asset_in=${baseDenom}&asset_out=${quoteDenom}&range=${tf}&asset_type=symbol`
+      // `/pairs/v1/historical/${poolId}/chart?asset_in=${baseDenom}&asset_out=${quoteDenom}&range=${tf}&asset_type=symbol`
+    );
+    console.log(baseURL);
+    makeObservable(this);
+  }
+
+  protected canFetch(): boolean {
+    return (
+      this.poolId !== "" &&
+      AvailableRangeValues.includes(this.tf) &&
+      this.tf != null &&
+      this.baseDenom != null &&
+      this.quoteDenom != null
+    );
+  }
+
+  @computed
+  get getChartPrices(): TokenPairHistoricalPrice[] | undefined {
```

Comment:
IMO it'd be cleaner to return a `[]` instead of `undefined`.

We can check if data is loading by using the base class' `isFetching` member